clippy_lints/src/unused_trait_names.rs

```diff
@@ -73,6 +73,8 @@ impl<'tcx> LateLintPass<'tcx> for UnusedTraitNames {
             && let Some(snip) = snippet_opt(cx, last_segment.ident.span)
             && self.msrv.meets(cx, msrvs::UNDERSCORE_IMPORTS)
             && !is_from_proc_macro(cx, &last_segment.ident)
+            // Skip if the ident of `as` is from a macro's input
```

Comment:
The comment does not seem to correspond to the test which also checks the last segment.